agents/cdu/fence_cdu.py

```diff
@@ -0,0 +1,143 @@
+    try:
+        (_, status) = outlets[options["--plug"]]
+        return status.lower().strip()
+    except KeyError:
+        fail(EC_STATUS)
+
+def set_power_status(conn, options):
+        outlets = {}
+        action = { 'on' : "on", 'off': "off" }[options["--action"]]
+        try:
+            conn.send("LIST OUTLETS\r\n")
+            conn.log_expect(options["--command-prompt"], int(options["--shell-timeout"]))
+            lines = conn.before.split("\n")
+            #    .A12     TowerA_Outlet12
+            #    .A12     test-01
+            show_re = re.compile('(\.\w+)\s+(\w+|\w+\W\w+)\s+')
+            for line in lines:
+                res = show_re.search(line)
+                if res != None:
+                    outlets[res.group(2)] = (res.group(1))
+            conn.send(action + " " + outlets[options["--plug"]] + "\r\n")
+            conn.log_expect(options["--command-prompt"], int(options["--shell-timeout"]))
+        except pexpect.EOF:
+            fail(EC_CONNECTION_LOST)
+        except pexpect.TIMEOUT:
+            fail(EC_TIMED_OUT)
+
+def main():
+    device_opt = [  "help", "version", "agent", "quiet", "verbose", "debug",
+                    "action", "ipaddr", "login", "passwd", "passwd_script",
+                    "secure", "port", "identity_file", "switch", "separator",
+                    "inet4_only", "inet6_only", "ipport", "telnet",
+                    "power_timeout", "shell_timeout", "login_timeout", "power_wait" ]
+
+    atexit.register(atexit_handler)
+
+    options = check_input(device_opt, process_input(device_opt))
+
+    ##
+    ## Fence agent specific defaults
+    #####
+    options["ssh_options"] = "-1 -c blowfish"
+    options["--command-prompt"] = "Switched CDU:"
+
+    docs = { }
+    docs["shortdesc"] = "Fence agent for a Sentry Switch CDU over telnet/ssh"
+    docs["longdesc"] = "fence_cdu is an I/O Fencing agent \
+which can be used with the Sentry Switch CDU. It logs into the device \
+via telnet/ssh and power's on/off an outlet."
+    docs["vendorurl"] = "http://www.servertech.com"
+    show_docs(options, docs)
+
+    ## Support for --plug [switch]:[plug] notation that was used before
+    opt_n = options.get("--plug")
+    if opt_n and (-1 != opt_n.find(":")):
+        (switch, plug) = opt_n.split(":", 1)
+        options["--switch"] = switch;
+        options["--plug"] = plug;
+
+    ##
+    ## Operate the fencing device
+    ####
+    conn = fence_login(options)
+    # disable output paging
+    conn.sendline("set option more disabled")
+    conn.log_expect(options["--command-prompt"], int(options["--login-timeout"]))
+    result = fence_action(conn, options, set_power_status, get_power_status, get_power_status)
+    ##
+    ## Logout from system
+    ##
+    ## In some special unspecified cases it is possible that
+    ## connection will be closed before we run close(). This is not
+    ## a problem because everything is checked before.
+    ######
+    try:
```

Comment:
You should move this to a function, and add it as an exit handler:
https://github.com/ClusterLabs/fence-agents/blob/8d746be92f191aa289f13a3703031c122a5e6cf3/agents/vmware_rest/fence_vmware_rest.py#L222